insights/core/__init__.py

```diff
@@ -426,6 +427,120 @@ def _scan(self):
 
         cls.scan(result_key, _scan)
 
+    def _get_after(self, timestamp, time_format='%Y-%m-%d %H:%M:%S', lines=None):
```

Comment:
Since you already provide a default time format, I think that you could just implement in child parsers where its different.  I think you might want to move the args around so that the least changed is last.  I don't think that rules would have to deal with the format.